core/dbt/parser/schemas.py

```diff
@@ -881,15 +884,31 @@ def parse_patch(self, block: TargetBlock[NodeTarget], refs: ParserRef) -> None:
                 f"Unexpected yaml_key {patch.yaml_key} for patch in "
                 f"file {source_file.path.original_file_path}"
             )
+        # handle disabled nodes
         if unique_id is None:
             # Node might be disabled. Following call returns list of matching disabled nodes
             found_nodes = self.manifest.disabled_lookup.find(patch.name, patch.package_name)
             if found_nodes:
-                # There might be multiple disabled nodes for this model
+                if len(found_nodes) > 1 and patch.config.get("enabled"):
+                    # There are multiple disabled nodes for this model and the schema file wants to enable one.
+                    # We have no way to know which one to enable.
+                    msg = (
+                        f"Found {len(found_nodes)} matching disabled nodes for '{patch.name}'. "
+                        "Multiple nodes for the same unique id cannot be disabled in the schema "
+                        "file. They must be disabled in `dbt_project.yml` or in the sql files."
+                    )
+                    raise ParsingException(msg)
```

Comment:
@jtcohen6 can I get some input on the error message here, please?

Reply:
Thanks for tagging me in — this looks mostly good to me!

Could we include the resource type? Is that available from the `patch`?

How about:
```
Found {len(found_nodes)} matching disabled nodes for {patch.resource_type} '{patch.name}'.
Multiple nodes with the same unique_id cannot be disabled in yaml resource properties.

If you need to have multiple disabled nodes with the same names, you should instead disable them
using in-file config, or resource-path config in `dbt_project.yml`.
```